scapy/fields.py

```diff
@@ -1726,8 +1726,8 @@ def __init__(self, name, default, enum, fmt="H"):
                 keys = enum.keys()
             else:
                 keys = list(enum)
-            if any(isinstance(x, str) for x in keys):
-                i2s, s2i = s2i, i2s
+                if any(isinstance(x, str) for x in keys):
```

Comment:
Why is this block now shifted under the `else` clause?

Reply:
If:
- the first if is followed: `keys` will be a list of integers -> not needed
- the second list is followed: we get the keys from `DADict.keys()` -> not needed